### PR TITLE
WT-4241 GNU-stack section should never be conditionally compiled out

### PR DESCRIPTION
**Pull request by keithbostic**

### CODE REVIEW

**Comment by keithbostic** [comment]
autobuild: retest this please

**Comment by agorrod** [comment]
lgtm - please merge if platforms team confirms this fixes their issue.

File: src/checksum/power8/crc32.sx [hunk 1]
```diff
@@ -775,11 +775,13 @@ FUNC_START(__crc32_vpmsum)
...
+#endif
 
 /*
  * Make sure the stack isn't executable with GCC (regardless of platform).
+ * Note: this section must always be output and can never be #ifdef'd out as
+ * part of conditional compilation.
  */
 #ifdef __ELF__
 .section	.note.GNU-stack,"",@progbits
 #endif
```
Comment (acmorrow) [inline src/checksum/power8/crc32.sx]: Do you want to add a scary
/* DO NOT SAY #ifndef BELOW THIS LINE, per above comment */

The comment you have is good, but will anyone read it before slapping an #endif on the last line of the file?

File: src/checksum/zseries/crc32le-vx.sx [hunk 2]
```diff
@@ -279,11 +279,13 @@ crc32_le_vgfm_generic:
...
 	lmg     %r14,%r15,240(%r15)
 	br	%r14
+#endif
 
 /*
  * Make sure the stack isn't executable with GCC (regardless of platform).
+ * Note: this section must always be output and can never be #ifdef'd out as
+ * part of conditional compilation.
  */
 #ifndef __clang__
```
Comment (acmorrow) [inline src/checksum/zseries/crc32le-vx.sx]: Not to be addressed in this review, but I wonder a bit. This file makes the GNU-stack note conditional on not using clang, the other makes it conditional on using ELF. I wonder which is right. Something for another day.

**Comment by acmorrow** [comment]
Note that this should be backported to all branches where WT-4198 was fixed (https://github.com/wiredtiger/wiredtiger/commit/96f004261d6336f81fbd0bc245e8aa07e7cc24ac)

File: src/checksum/power8/crc32.sx (hunk 1 above)
Comment (keithbostic) [reply src/checksum/power8/crc32.sx]: Will do.

File: src/checksum/zseries/crc32le-vx.sx (hunk 2 above)
Comment (keithbostic) [reply src/checksum/zseries/crc32le-vx.sx]: Good catch, thanks -- I'll use `__ELF__`, afaict, that's accepted practice.

**Pull request merged**